lib/serum/header_parser.ex

```diff
@@ -61,16 +60,27 @@ defmodule Serum.HeaderParser do
   def parse_header(data, options, required \\ []) do
     case extract_header(data, [], false) do
       {:ok, header_lines, rest_data} ->
-        key_strings = options |> Keyword.keys() |> Enum.map(&Atom.to_string/1)
+        key_strings = options |> Keyword.keys() |> Enum.map(&to_string/1)
+        req_strings = Enum.map(required, &to_string/1)
 
-        kv_list =
+        kv_lists =
           header_lines
           |> Enum.map(&split_kv/1)
-          |> Enum.filter(fn {k, _} -> k in key_strings end)
+          |> Enum.group_by(&(elem(&1, 0) in key_strings))
 
-        with [] <- find_missing(kv_list, required),
-             {:ok, new_kv} <- transform_values(kv_list, options, []) do
-          {:ok, {Map.new(new_kv), rest_data}}
+        %{
+          true: accepted_kv,
+          false: extra_kv
+        } = Map.merge(%{true: [], false: []}, kv_lists)
+
+        with [] <- find_missing(accepted_kv, req_strings),
+             {:ok, parsed} <- transform_values(accepted_kv, options, []) do
+          extras =
+            Enum.map(extra_kv, fn {k, v} ->
+              {k, ValueTransformer.transform_value(k, v, :string)}
+            end)
+
+          {:ok, {Map.new(parsed), Map.new(extras), rest_data}}
```

Comment:
_tl;dr: Please see the source code for [`HeaderParser.extract_header/3`](https://github.com/Dalgona/Serum/blob/c8c6a788a086182cfd8485d7daa1d4142d6feba0/lib/serum/header_parser.ex#L93-L120)._

`rest_data` in that line refers to any trailing data after the header area. In case of a blog post, for example, that should be the body of a blog post (yet unprocessed).

Since `Serum.HeaderParser.parse_header/3` takes the whole text from a file, and it does not know the positions where the header starts and ends, this function reads the text line by line, _until_ it identifies the complete header area. Any trailing data not read becomes `rest_data`, and it will be passed to the next processing step as is.